mysql-test/suite/innodb/r/mem_pressure.result

```diff
@@ -0,0 +1,23 @@
+#
+# MDEV-24670 avoid OOM by linux kernel co-operative memory management
+#
+CREATE TABLE t1 (t TEXT) ENGINE=InnoDB;
+INSERT INTO t1 SELECT CONCAT(REPEAT('junk text', 500), CAST(seq AS CHAR)) FROM seq_1_to_1000;
+SELECT CAST(VARIABLE_VALUE AS INTEGER) INTO @dirty_prev FROM INFORMATION_SCHEMA.GLOBAL_STATUS WHERE VARIABLE_NAME='Innodb_buffer_pool_pages_dirty';
+SELECT CAST(VARIABLE_VALUE AS INTEGER) INTO @flush_prev FROM INFORMATION_SCHEMA.GLOBAL_STATUS WHERE VARIABLE_NAME='Innodb_buffer_pool_pages_flushed';
+set debug_dbug="d,trigger_garbage_collection";
+SET GLOBAL innodb_buffer_pool_size=@@innodb_buffer_pool_size;
+SELECT SLEEP(5);
+SLEEP(5)
+0
```

Comment:
Instead of sleeping in the test, I think that it would be better to revise the debug function `mem_pressure::trigger_collection()` so that it will poll until the garbage collection has been invoked. That could make use of a debug counter.

Reply:
The entire garbage collection is performed in the same thread as 'SET GLOBAL innodb_buffer_pool_size=@@innodb_buffer_pool_size;'. As its not a background task, no `SLEEP` or addition synchronization is required.